consensus/src/dag/dag_handler.rs

```diff
@@ -91,51 +106,95 @@ impl NetworkHandler {
                 (msg, epoch, author, responder) = verified_msg_stream.select_next_some() => {
-                    monitor!("dag_on_verified_msg", match self.process_verified_message(msg, epoch, author, responder).await {
-                        Ok(sync_status) => {
-                            if matches!(sync_status, SyncOutcome::NeedsSync(_) | SyncOutcome::EpochEnds) {
-                                return sync_status;
-                            }
-                        },
-                        Err(e) =>  {
-                            warn!(error = ?e, "error processing rpc");
-                        }
-                    });
+                    let verified_msg_processor = verified_msg_processor.clone();
+                    let f = executor.spawn(async move {
+                        monitor!("dag_on_verified_msg", {
+                            match verified_msg_processor.process_verified_message(msg, epoch, author, responder).await {
+                                Ok(sync_status) => {
+                                    if matches!(
+                                        sync_status,
+                                        SyncOutcome::NeedsSync(_) | SyncOutcome::EpochEnds
+                                    ) {
+                                        return Some(sync_status);
+                                    }
+                                },
+                                Err(e) => {
+                                    warn!(error = ?e, "error processing rpc");
+                                },
+                            };
+                            None
+                        })
+                    }).await;
+                    futures.push(f);
                 },
-                Some(new_round) = self.new_round_event.recv() => {
-                    monitor!("dag_on_new_round_event", {
-                        self.dag_driver.enter_new_round(new_round).await;
-                        self.node_receiver.gc();
-                    });
-                }
-                Some(res) = self.node_fetch_waiter.next() => {
-                    monitor!("dag_on_node_fetch", match res {
-                        Ok(node) => if let Err(e) = self.node_receiver.process(node).await {
-                            warn!(error = ?e, "error processing node fetch notification");
-                        },
-                        Err(e) => {
-                            debug!("sender dropped channel: {}", e);
-                        },
-                    });
+                Some(status) = futures.next() => {
+                    if let Some(status) = status.expect("must finish") {
+                        return status;
+                    }
+                },
+                Some(new_round) = new_round_event.recv() => {
+                    let dag_driver_clone = dag_driver.clone();
+                    let node_receiver_clone = node_receiver.clone();
+                    executor.spawn(async move {
```

Comment:
commented on previous pr, I think this can be synchronous assuming nothing is blocking?